### ISSUE TITLE
Tab component throws warning about Href and tabIndex

### ISSUE THREAD

**Issue by amortiz** [carbon-design-system/carbon#7237]
Having this code:
`<Tabs> <Tab> hello </Tab> </Tabs>` throws console warnings, which break our unit tests:

```
index.js:27 Warning: The prop `href` has been deprecated for the Tab component. It will be removed in the next major release 
    in App (at src/​index.js:13)
index.js:27 Warning: The prop `tabIndex` has been deprecated for the Tab component. It will be removed in the next major release 
```
I'm not passing those in, probably the default props inside Tab is doing this?

You can see here
https://codesandbox.io/s/sleepy-sid-h56nf

**Comment by tw15egan** [comment]
Fixed in https://github.com/carbon-design-system/carbon/pull/7221 and  https://github.com/carbon-design-system/carbon/pull/7231 and will be included in the next release 👍

**Comment by starpit** [comment]
Will this fix be included in a 7.23.x? (we can't afford to ride your wave of minor releases, as they consistently introduce breaking changes and regressions)
